src/filebrowser/file-browser-dialog.cpp

```diff
@@ -794,70 +833,87 @@ void FileBrowserDialog::onDownloadFinished(bool success)
+//             forceRefresh();
+// 
+//         if (task->error() == FileNetworkTask::TaskCanceled)
+//             return;
+// 
+//         QString msg = tr("Failed to upload file: %1").arg(task->errorString());
+//         seafApplet->warningBox(msg, this);
+//         return;
+//     }
+// 
+//     QString local_path = task->localFilePath();
+//     QStringList names;
+// 
+//     // Upload Directory Task
+//     if (qobject_cast<FileUploadDirectoryTask *>(sender())) {
+//         const SeafDirent dir = SeafDirent::dir(task->name());
+//         // TODO: insert the Item prior to the item where uploading occurs
+//         table_model_->insertItem(0, dir);
+//         return;
+//     }
+// 
+//     // Upload Multiple Task
+//     FileUploadMultipleTask *multi_task = qobject_cast<FileUploadMultipleTask *>(sender());
+// 
+//     if (multi_task == NULL) {
+//       names.push_back(task->name());
+//       local_path = QFileInfo(local_path).absolutePath();
+// =======
+    SeafDirent dirent;
+    if (is_dir) {
+        dirent = SeafDirent::dir(name);
+// >>>>>>> temp commit for rebase.
     } else {
-      names = multi_task->names();
-      local_path = QFileInfo(local_path).absoluteFilePath();
-    }
-
-    // require a forceRefresh if conflicting filename found
-    Q_FOREACH(const QString &name, names)
-    {
-        if (findConflict(name, table_model_->dirents())) {
-            forceRefresh();
-            return;
-        }
+        dirent = SeafDirent::file(name, 0);
     }
 
     // add the items to tableview
-    Q_FOREACH(const QString &name, names) {
-        const QFileInfo file = QDir(local_path).filePath(name);
-        const SeafDirent dirent = SeafDirent::file(name, static_cast<quint64>(file.size()));
-        if (task->useUpload())
-            table_model_->appendItem(dirent);
-        else
-            table_model_->replaceItem(name, dirent);
-    }
+// <<<<<<< 1d9fda7ec5883b1c275edb215721bdb51b45d8d8
+//     Q_FOREACH(const QString &name, names) {
+//         const QFileInfo file = QDir(local_path).filePath(name);
+//         const SeafDirent dirent = SeafDirent::file(name, static_cast<quint64>(file.size()));
+//         if (task->useUpload())
+//             table_model_->appendItem(dirent);
+//         else
+//             table_model_->replaceItem(name, dirent);
+//     }
+// =======
+    table_model_->appendItem(dirent);
+}
+
+void FileBrowserDialog::onUploadSuccess(const FileTaskRecord* task)
+{
+    QString local_path = task->local_path;
+    const QFileInfo file = QFileInfo(local_path);
+    const QString file_name = file.fileName();
+    const SeafDirent dirent = SeafDirent::file(file_name, static_cast<quint64>(file.size()));
+    table_model_->replaceItem(file_name, dirent);
+}
+
+void FileBrowserDialog::onUploadFailed(const FileTaskRecord* task)
+{
+    QString msg = tr("Failed to upload file: %1").arg(task->fail_reason);
+    seafApplet->warningBox(msg, this);
```

Comment:
If the upload is canceled by the user, then we should not show a "task cancelled" error dialog.